src/lexer.cpp

```diff
@@ -2,4 +2,165 @@
 // Created by moldo on 06/03/2021.
 //
 
+
+#include <string>
+#include <algorithm>
 #include "lexer.h"
+#include "defs.h"
+
+#define XASMEOFConstant 3
+
+Lexer::Lexer(std::string &source) {
+        this->source = source;
+        position = 0;
+        currentChar = 0;
+
+        // Convert source to lowercase
+        for (char &c : this->source)
+                if (std::isupper(c))
+                        c = (char)std::tolower(c);
+
+        // Initialize currentChar with first character in source
+        nextChar();
+}
+
+Token Lexer::nextToken() {
+        Token t {};
+
+        skipSpaces();
+
+        t.value += currentChar;
+
+        switch(currentChar) {
+                case '\r': case '\n': case '\f':
+                        t.type = TokenType::NewLine;
+
+                        while (isNewLine() || isSpace())
+                                nextChar();
+
+                        break;
+                case '(':
+                        t.type = TokenType::Lparan;
+                        nextChar();
+
+                        break;
+                case ')':
+                        t.type = TokenType::Rparan;
+                        nextChar();
+
+                        break;
+                case '.':
+                        t.type = TokenType::Dot;
+                        nextChar();
+
+                        break;
+                case ':':
+                        t.type = TokenType::Colon;
+                        nextChar();
+
+                        break;
+                case ',':
+                        t.type = TokenType::Comma;
+                        nextChar();
+
+                        break;
+                case ';':
+                        t.type = TokenType::Comment;
+                        nextChar();
+
+                        while(!isNewLine()) {
+                                t.value += currentChar;
+                                nextChar();
+                        }
+
+                        break;
+                case '0' ... '9':
+                case '-':
+                        t.type = TokenType::Number;
+                        nextChar();
+
+                        while (std::isdigit(currentChar)) {
+                                t.value += currentChar;
+                                nextChar();
+                        }
+
+                        break;
+                case XASMEOFConstant: case 0:
```

Comment:
Please clarify what XASMEOF with and without constant are. I don't quite get it from  the code. Eventually add a comment.